emperor/support_files/js/opacity-view-controller.js

```diff
@@ -52,10 +52,21 @@ define([
 
     // webgl acts up with transparent objects, so we only set them to be
     // explicitly transparent if the opacity is not at full
-    _.each(group, function(element) {
-      scope.markers[element.idx].material.transparent = transparent;
-      scope.markers[element.idx].material.opacity = opacity;
-    });
+    if (scope.decomp.isScatterType()) {
+      _.each(group, function(element) {
+        scope.markers[element.idx].material.transparent = transparent;
+        scope.markers[element.idx].material.opacity = opacity;
+      });
+    }
+    else if (scope.decomp.isArrowType()) {
+      _.each(group, function(element) {
```

Comment:
Would it make sense to pull these anonymous functions out? It has the benefit of simplifying the if/elseif by reducing the LOC of each branch

Reply:
This is a great suggestion. I think the code is much more readable now 👍